odoo/orm/models.py

```diff
@@ -5575,23 +5575,23 @@ def with_context(self, *args, **kwargs) -> Self:
 
             The returned recordset has the same prefetch object as ``self``.
         """  # noqa: RST210
-        if (args and 'force_company' in args[0]) or 'force_company' in kwargs:
+        context = dict(ctx if ctx is not None else self.env.context, **kwargs)
+        if 'force_company' in context:
```

Comment:
IMO, the goal to put that before was to warn only when `force_company` is actually updated, not when it comes from the `self.env.context`.

Note that the `force_company` warning can be removed (maybe) IMO.

Reply:
Maybe these warnings could become deprecation warnings (and as such be shown only once).